packages/astro/src/cli/add.ts

```diff
@@ -0,0 +1,345 @@
+import type yargs from 'yargs-parser';
+import path from 'path';
+import fs from 'fs/promises';
+import { exec } from 'child_process';
+import { fileURLToPath } from 'url';
+import { diffLines } from 'diff';
+import boxen from 'boxen';
+import prompts from 'prompts';
+import preferredPM from 'preferred-pm';
+import { resolveConfigURL } from '../core/config.js';
+import { apply as applyPolyfill } from '../core/polyfill.js';
+import { error, info, debug, LogOptions } from '../core/logger.js';
+import * as msg from '../core/messages.js';
+import { dim, red, cyan, green, bold, magenta } from 'kleur/colors';
+import { parseNpmName } from '../core/util.js';
+import { t, parse, visit, ensureImport, wrapDefaultExport, generate } from '../transform/index.js';
+
+export interface AddOptions {
+	logging: LogOptions;
+	cwd?: string;
+	flags: yargs.Arguments;
+}
+
+export interface IntegrationInfo {
+	id: string;
+	packageName: string;
+	dependencies: [name: string, version: string][];
+}
+
+const DEFAULT_CONFIG_STUB = `import { defineConfig } from 'astro/config';\n\nexport default defineConfig({});`;
+
+export async function add(names: string[], { cwd, flags, logging }: AddOptions) {
+	if (names.length === 0) {
+		error(logging, null, `\n${red('No integration specified!')}\n${dim('Try using')} astro add ${cyan('[name]')}`);
+		return;
+	}
+	const root = cwd ? path.resolve(cwd) : process.cwd();
+	let configURL = await resolveConfigURL({ cwd, flags });
+	applyPolyfill();
+	if (configURL) {
+		debug('add', `Found config at ${configURL}`);
+	} else {
+		info(logging, 'add', `Unable to locate a config file, generating one for you.`);
+		configURL = new URL('./astro.config.mjs', `file://${root}/`);
+		await fs.writeFile(fileURLToPath(configURL), DEFAULT_CONFIG_STUB, { encoding: 'utf-8' });
+	}
+
+	const integrations = await validateIntegrations(names);
+
+	// Add integrations to astro config
+	// TODO: At the moment, nearly nothing throws an error. We need more errors!
+	let ast: t.File | null = null;
+	try {
+		ast = await parseAstroConfig(configURL);
+
+		debug('add', 'Parsed astro config');
+
+		const defineConfig = t.identifier('defineConfig');
+		ensureImport(ast, t.importDeclaration([t.importSpecifier(defineConfig, defineConfig)], t.stringLiteral('astro/config')));
+		wrapDefaultExport(ast, defineConfig);
+
+		debug('add', 'Astro config ensured `defineConfig`');
```

Comment:
nice! I was actually going to suggest that it would make sense to limit our launch support to only config files with a single `export default defineConfig()`. Looks like you've already done that here!